internal/component/otelcol/exporter/awss3/awss3.go

```diff
@@ -155,6 +155,10 @@ func (args *S3Uploader) Convert() awss3exporter.S3UploaderConfig {
 		Compression:       args.Compression,
 		ACL:               args.ACL,
 		StorageClass:      args.StorageClass,
+		// todo: allow configuring retries https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39509
+		RetryMode:        awss3exporter.DefaultRetryMode,
+		RetryMaxAttempts: awss3exporter.DefaultRetryMaxAttempts,
+		RetryMaxBackoff:  awss3exporter.DefaultRetryMaxBackoff,
```

Comment:
We will want to expose this in Alloy config and add to documentation. 

Similarly, we'll need to go through changelog and ensure we expose all the new functionality or describe any breaking changes.

Reply:
Cool, opened this one: https://github.com/grafana/alloy/issues/3844